Tools/Scripts/test262-helpers/test262-runner.pl

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env perl
+
+# Copyright (C) 2018 Bocoup LLC. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of source code must retain the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer.
+# 2. Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials
+#    provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+# OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+# TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+use strict;
+use warnings;
+use 5.8.8;
+
+use File::Find;
+use File::Temp qw(tempfile tempdir);
+use File::Spec::Functions qw(abs2rel);
+use Cwd 'abs_path';
+use FindBin;
+
+######
+# # Use the following code run this script directly from Perl.
+# # Otherwise, just use carton.
+use Config;
+use Encode;
+
+BEGIN {
+    $ENV{DBIC_OVERWRITE_HELPER_METHODS_OK} = 1;
+
+    unshift @INC, ".";
+    unshift @INC, "$FindBin::Bin/lib";
+    unshift @INC, "$FindBin::Bin/local/lib/perl5";
+    unshift @INC, "$FindBin::Bin/local/lib/perl5/$Config{archname}";
+
+    $ENV{LOAD_ROUTES} = 1;
+}
+######
+
+use YAML qw(Load);
+use Parallel::ForkManager;
+use Getopt::Long qw(GetOptions);
+use Pod::Usage;
+
+# Commandline args
+my $cliProcesses;
+my @cliTestDirs;
+my $JSC;
+
+processCLI();
+
+my $tempdir = tempdir();
+
+my $test262Dir = abs_path("$FindBin::Bin/../../../JSTests/test262");
+my $harnessDir = "$test262Dir/harness";
+
+my @default_harnesses = (
+    "$harnessDir/sta.js",
+    "$harnessDir/assert.js",
+    "$harnessDir/doneprintHandle.js",
+    "$FindBin::Bin/agent.js"
+);
+
+my $tests_log = "$FindBin::Bin/tests.log";
+
+# TODO: derive this number by probing the system.
+my $cpus = 8;
+my $max_process = $cpus * 8;
+my $pm = Parallel::ForkManager->new($max_process);
+my @files;
+my ($resfh, $resfilename) = getTempFile();
+
+my ($deffh, $deffile) = getTempFile();
+print $deffh getHarness(<@default_harnesses>);
+
+my $startTime = time();
+
+main();
+
+sub processCLI {
+    my $help = 0;
+
+    GetOptions(
+        'j|jsc=s' => \$JSC,
+        't|t262=s@' => \@cliTestDirs,
```

Comment:
we need to break this one. 

We need to set a test262Dir which is a central point to locate test262, so we should use only 1 argument for this.

Also, specific paths to run should be the default argument, no need to use a flag. e.g:

```shell
# setting the location of test262
./test262-runner.pl -t262=../../../JSTests/test262

# setting the paths to run (we might use globs)
./test262-runner.pl test/built-ins/Array test/built-ins/TypedArray*

# custom location and paths
./test262-runner.pl -t262=~/dev/test262 test/built-ins/Array test/built-ins/TypedArrays
```